creating_gs_app.md

```diff
@@ -29,46 +29,76 @@ to put it, we have created a `guest_science_projects` folder for you to use.
-[`freeflyer/simulation/sim_overview.md`](https://github.com/nasa/astrobee/blob/master/simulation/sim_overview.md#setting-up-your-environment).
+[`astrobee/simulation/running_the_sim.md`](https://github.com/nasa/astrobee/blob/master/simulation/running_the_sim.md).
 
 ## Build the Astrobee API
 
 ### Generating ARS ROS Messages (`ff_msgs`)
 
-Ensure `rosjava` is installed:
-
-    you@machine:~ $ sudo apt-get install ros-kinetic-rosjava
+**Ubuntu 16.04**
 
+Ensure `rosjava` is installed:
+```shell
+  sudo apt-get install ros-kinetic-rosjava
+```
 Build the ff_msgs jar
-
-    you@machine:~ $ cd $BUILD_PATH
-    you@machine:native $ make rebuild_cache
-    you@machine:native $ make ff_msgs_generate_messages_java_gradle
-
+ <!--- - Note: You may not need to build the ff_msgs jar, as they should be built when you install the Astrobee Software. In the event that they weren't, run the commands below. -->
+ 
+ <!--- This is something I was told, and so not to mislead people I will leave it out -->
+ 
+```shell
+  cd $BUILD_PATH
+  make rebuild_cache
+  make ff_msgs_generate_messages_java_gradle
+```
+**Ubuntu 20.04**
+
+As there is no ros-noetic-java for Ubuntu 20.04, the current solution is for
+developers to download the jar files here:
+
+https://github.com/nasa/astrobee/actions/workflows/msgs_jar.yaml?query=branch%3Adevelop
+ - Note: Be sure to download the most recent one that is passing.
+  
 ### Letting gradle access ARS messages
 
 The astrobee_api project is configured to look at the local maven repository
 to access the `ff_msgs` jar. The default location for this repository is:
-`$HOME/.m2/repository`. If you are building this project on the same machine
-that you build flight software on, the easiest thing to do is symlink the
+`$HOME/.m2/repository`. 
+```shell
+  mkdir -p $HOME/.m2
+  cd $HOME/.m2
+```
+**Ubuntu 16.04**
+
+If you are building this project on the same machine
+that you built the astrobeeunseen comment in github software on, the easiest thing to do is symlink the
 rosjava generated files into the right location:
-
-    you@machine:~ $ mkdir -p $HOME/.m2
-    you@machine:.m2 $ cd $HOME/.m2
-    you@machine:.m2 $ ln -s $BUILD_PATH/devel/share/maven repository
-
+```shell
+  ln -s $BUILD_PATH/share/maven repository
+```
 Otherwise you will have to copy the contents of the maven directory out
-of `devel/shared` into `$HOME/.m2/repository`.
-
-### Building the JAR Files
+of `shared` into `$HOME/.m2/repository`.
 
-    you@machine:~ $ cd $ANDROID_PATH/astrobee_api
-    you@machine:astrobee_api $ ./gradlew build
+**Ubuntu 20.04**
 
+For developers running Ubuntu 20.04, `maven` will not exist in `$BUILD_PATH/share`. 
+And so, the easiest thing to do is to create a subdirectory called `repository` in `.m2`, simply to house the jar files.
+```shell
+  mkdir repository
+```
+Once the `repository` directory has been created, simply move the jar files into it from wherever they are located. 
```

Comment:
Do you think we need more specifics? I suppose they need to copy the whole folder structure for example, or does it work with the jar files only?

Reply:
I can make it more specific, I kind of assumed that either they see the error and it will tell them the path that it is looking for which if I remember correctly was .m2/repository/org/.. and so they know to add the whole folder, or they would just add the whole folder to begin with and not the files only. 

But I see your point, I can add that detail to clarify.